native/src/main/java/io/ballerina/lib/ldap/Ldap.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.ldap;
+
+import com.unboundid.ldap.sdk.Attribute;
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPException;
+import com.unboundid.ldap.sdk.LDAPResult;
+import com.unboundid.ldap.sdk.Modification;
+import com.unboundid.ldap.sdk.ModificationType;
+import com.unboundid.ldap.sdk.ModifyRequest;
+import com.unboundid.ldap.sdk.SearchResultEntry;
+import com.unboundid.util.Base64;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.ValueUtils;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.unboundid.ldap.sdk.ResultCode.NO_SUCH_OBJECT;
+import static io.ballerina.lib.ldap.ModuleUtils.LDAP_RESPONSE;
+import static io.ballerina.lib.ldap.ModuleUtils.MATCHED_DN;
+import static io.ballerina.lib.ldap.ModuleUtils.OBJECT_GUID;
+import static io.ballerina.lib.ldap.ModuleUtils.OBJECT_SID;
+import static io.ballerina.lib.ldap.ModuleUtils.OPERATION_TYPE;
+import static io.ballerina.lib.ldap.ModuleUtils.RESULT_STATUS;
+import static io.ballerina.lib.ldap.Utils.ENTRY_NOT_FOUND;
+
+/**
+ * This class handles APIs of the LDAP client.
+ */
+public class Ldap {
+    private Ldap() {
+    }
+
+    public static BError initLdapConnection(BObject ldapClient, BMap<BString, Object> config) {
```

Comment:
```suggestion
    public static Object initLdapConnection(BObject ldapClient, BMap<BString, Object> config) {
```

Reply:
Decided to keep the `BError` type as it also can be used here as the return type